internal/querynodev2/segments/manager.go

```diff
@@ -313,58 +315,57 @@ func (segments segments) RangeWithFilter(criterion *segmentCriterion, process fu
 		return
 	}
 
-	var candidates []map[int64]Segment
+	var candidates []*typeutil.ConcurrentMap[typeutil.UniqueID, Segment]
```

Comment:
Why we need concurrentMap here since its scope is just this function

Reply:
Since `candidates` are directly constructed from `growingSegments` and `sealedSegments`, which are of the `ConcurrentMap` type, using a regular map for `candidates` would require a copy operation, making it inefficient.

Reply:
Actually it is shallow copy so not that heavy. But I don't think it is something have to be done.